net: lib: coap: Service support (#64265)
zephyrproject-rtos/zephyr: samples/net/sockets/coap_server/README.rst

.. zephyr:code-sample:: coap-server
   :name: CoAP service

>    :relevant-api: coap udp

Reviewer: ```suggestion
   :relevant-api: coap coap_service udp
```

>    Use the CoAP server subsystem that exposes CoAP resources.

Reviewer: better wording below?

```suggestion
   Use the CoAP server subsystem to register CoAP resources.
```

Reviewer: It feels like the rest of the README could be slightly reworded too and provide a brief explanation of how it now uses the CoAP service?
